### PR TITLE
SIMSBIOHUB-442: Add support for hiding/showing table columns for Obserations and Telemetry

### PR DESCRIPTION
**Pull request by curtisupshall**
## Links to Jira Tickets

 - https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-442

## Description of Changes

- Add support for hiding/showing table columns for Obserations and Telemetry
- Added `sessionStorage` call in order to cache column visibility across app reloads
- Renamed `ObservationComponent` and `ManualTelemetryComponent` to `ObservationsTableContainer` and `ManualTelemetryTableContainer`

## Testing Notes

- Visit the Manage Observations or Manage Telemetry page
- Click the cog icon
- You should be able to:
  - [ ] Show/hide all columns
  - [ ] Show/hide additional columns
 - Show/hide some or all columns, then refresh the page. You should:
   - [ ] See that the visiblity state has persisted across refreshes

### CODE REVIEW

**Comment by NickPhura** [comment]
Noticed a few things (not sure if these were there before or not):

On the Manage Observations page
- Click `Add Record`
- Leave some or all fields empty
- Click `Save`
- Validation banner is displayed
- Click `Discard Changes`
- Validation banner is still displayed

On the Manage Observations page
- Click `Add Record`
- Leave some or all fields empty
- Click `Save`
- Validation banner is displayed
- Hide one of the columns that is marked as in error
- Click the `Prev` or `Next` buttons on the validation banner.
- Error is thrown when it tries to navigate to one of the hidden columns
  - `visibleColumns[params.colIndex] is undefined`

There are also a couple of react errors in the console:
- When I first click `Add Record`. 
- When I first enter data in the `Date` column.

**Review by al-rosenthal — DISMISSED** [review]
Ran into a bug:

- Remove all columns from a table
- Add a new record
- Hit the save button
- Use the next/ prev buttons to navigate
- Error is thrown

File: app/src/components/data-grid/DataGridValidationAlert.tsx [hunk 1]
```diff
@@ -98,8 +98,14 @@ const DataGridValidationAlert = <RowType extends Record<any, any>>(props: IDataG
...
       const field = String(focusedError.field);
+      const isFieldVisible = props.muiDataGridApiRef.getVisibleColumns().some((column) => column.field === field);
+
+      if (!isFieldVisible) {
+        return;
+      }
+
       const rowIndex = props.muiDataGridApiRef.getSortedRowIds().indexOf(focusedError.rowId);
-      const colIndex = props.muiDataGridApiRef.getColumnIndex(field);
+      const colIndex = props.muiDataGridApiRef.getColumnIndexRelativeToVisibleColumns(field);
```
Comment (NickPhura) [inline app/src/components/data-grid/DataGridValidationAlert.tsx]: Nice find. 

Also, yet another situation where if you search "getColumnIndexRelativeToVisibleColumns" you get exactly 1 search result (from the MUI doc). Crazy.

Comment (NickPhura) [reply app/src/components/data-grid/DataGridValidationAlert.tsx]: Hmm, I'm still getting an error when I click through the banner errors:

```
visibleColumns[params.colIndex] is undefined
./node_modules/@mui/x-data-grid/hooks/features/scroll/useGridScroll.js/useGridScroll/scrollToIndexes<@http://localhost:7107/static/js/bundle.js:150323:9
./src/components/data-grid/DataGridValidationAlert.tsx/DataGridValidationAlert/focusErrorAtIndex<@http://localhost:7107/static/js/bundle.js:3164:29
./src/components/data-grid/DataGridValidationAlert.tsx/DataGridValidationAlert/handleNext</<@http://localhost:7107/static/js/bundle.js:3138:24
basicStateReducer@http://localhost:7107/static/js/bundle.js:266236:45
updateReducer@http://localhost:7107/static/js/bundle.js:266344:26
updateState@http://localhost:7107/static/js/bundle.js:266626:14
useState@http://localhost:7107/static/js/bundle.js:267327:20
useState@http://localhost:7107/static/js/bundle.js:287185:25
DataGridValidationAlert@http://localhost:7107/static/js/bundle.js:3100:76
```

Comment (NickPhura) [reply app/src/components/data-grid/DataGridValidationAlert.tsx]: Also, not sure if its critical right this second, but the validation banner error count is only considering the visible columns. So even if 12 columns are wrong, if only 5 are visible, it just shows '1/5' in the banner. Which, maybe is fine for now?

**Review by NickPhura — DISMISSED** [review]

File: app/src/components/data-grid/DataGridValidationAlert.tsx (hunk 1 above)
Comment (curtisupshall) [reply app/src/components/data-grid/DataGridValidationAlert.tsx]: > Also, not sure if its critical right this second, but the validation banner error count is only considering the visible columns. So even if 12 columns are wrong, if only 5 are visible, it just shows '1/5' in the banner. Which, maybe is fine for now?

I'm not able to reproduce this behaviour 🤔

**Review by al-rosenthal — APPROVED** [review]
ran this locally and it works as expected 🔧

**Review by NickPhura — APPROVED** [review]
🌔

**Pull request merged**